src/uwtools/resources/rocoto.jsonschema

```diff
@@ -1,12 +1,174 @@
+      "properties": {
+        "var": {
+          "$ref": "#/$defs/var"
         }
+      },
+      "patternProperties": {
+        "^task(_[a-z0-9_]+)?$": {
+          "$ref": "#/$defs/task"
+        },
+        "^metatask(_[a-z0-9_]+)?$": {
+          "$ref": "#/$defs/metatask"
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "var"
+      ],
+      "minProperties": 2,
+      "maxProperties": 3
+    },
+    "task": {
+      "type": "object",
+      "properties": {
+        "attrs": {
+            "maxtries": {
+                "type": "string",
+                "format": "number",
+                "minimum": 0
+            },
+            "throttle": {
+                "type": "integer",
+                "minimum": 0
+            },
+            "cycledefs": {
+                "type": "string"
+            },
+            "final": {
+                "type": "boolean"
+            }
+        },
+        "command": {
+          "type": "string"
+        },
+        "cores": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "nodes": {
+          "type": "string"
+        },
+        "native": {
+            "type": "string"
+        },
+        "walltime": {
+          "$ref": "#/$defs/timestr"
+        },
+        "account": {
+            "type": "string"
+        },
+        "jobname": {
+            "type": "string"
+        },
+        "queue": {
+            "type": "string"
+        },
+        "partition": {
+            "type": "string"
+        },
+        "memory": {
+            "type": "string"
+        },
+        "rewind": {
+            "type": "string"
+        },
+        "envar": {
+            "type": "object"
+        },
+        "shared": {
+            "type": "string"
+        },
+        "exclusive": {
+            "type": "string"
+        },
+        "stdout": {
+            "type": "string"
+        },
+        "stderr": {
+            "type": "string"
+        },
+        "join": {
+            "type": "string"
+        },
+        "deadline": {
+            "$ref": "#/$defs/datePattern"
+        }
+      },
+      "additionalProperties": false,
+      "required": [
+        "command",
+        "walltime"
+      ],
+      "oneOf": [
+        {
+            "required": ["cores"]
+        },
+        {
+            "required": ["nodes"]
+        },
+        {
+            "required": ["native"]
+        }
+      ],
+      "anyOf": [
+        {
+            "not": {
+                "required": ["shared"]
+            }
+        },
+        {
+            "not": {
+                "required": ["exclusive"]
+            }
+        }
+      ],
+      "anyOf":[
+        {
+            "not": {
+                "required": ["stdout"]
+            }
+        },
+        {
+            "not": {
+                "required": ["join"]
+            }
+        }
+      ],
+      "dependencies": {
+        "stdout": ["stderr"],
+        "stderr": ["stdout"]
+      }
```

Comment:
On page https://json-schema.org/understanding-json-schema/reference/conditionals, if you search for `dependencies`, it sounds like this syntax as semi-deprecated (or, at least, belongs to an older draft) in favor of two more-specific keywords, `dependentRequired` and `dependentSchemas`. Would it be useful / possible for us to "upgrade" to that newer syntax here?

Reply:
I attempted to do this by using the `dependentRequired` keyword and when including that over `dependencies` it automatically passes the tests when we should see a failure if both `stdout` and `stderr` are not supplied together. I am working on this issue, but unsure why `dependentRequired` would not behavior the same as `dependencies` since from the link you supplied they seem to be a 1:1 replacement for each other. I added another test for the `dependencies` feel free to take a look at it and test it out.

Reply:
I wouldn't spend a lot of time on it if it doesn't fall into place easily -- we should just go with what you've got working, and we can update later if we need to. If you don't like not having it behave the way you expect, maybe spend an hour or so on it before shelving it?

Reply:
I think the added test case on the dependencies will help catch any changes in the future on the issue. For example, if someone did change it to `dependentRequired` blindly without testing it, then the test case would catch this and it would fail. I'm happy leaving this how it is currently. The additions over the last two days have been monumental in getting this ticket to the point of where we wanted it to be at.